src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
It is the way it is because users specify a cooker when populating a config used to create a SharedRealm. This is why it is stored in the config. If we were able to hold onto a shared_ptr to the cooker in the `sync::Client` then I think there would no longer be an issue.